File: LeetCode/Math/1344-Angle-Between-Hands-of-a-Clock.js
/**
 * Problem: 1344. Angle Between Hands of a Clock
 * Platform: LeetCode
 * Difficulty: Medium
 *
 * Time Complexity: O(1)
 * Space Complexity: O(1)
 */

/**
 * @param {number} hour
 * @param {number} minutes
 * @return {number}
 */
var angleClock = function(hour, minutes) {
    const hourAngle = (hour % 12) * 30 + minutes * 0.5;
    const minuteAngle = minutes * 6;

    const diff = Math.abs(hourAngle - minuteAngle);

    return Math.min(diff, 360 - diff);
};

File: LeetCode/Math/1344-Angle-Between-Hands-of-a-Clock.py
"""
Problem: 1344. Angle Between Hands of a Clock
Platform: LeetCode
Difficulty: Medium

Time Complexity: O(1)
Space Complexity: O(1)
"""

class Solution:
    def angleClock(self, hour: int, minutes: int) -> float:
        hour_angle = (hour % 12) * 30 + minutes * 0.5
        minute_angle = minutes * 6

        diff = abs(hour_angle - minute_angle)

        return min(diff, 360 - diff)